tools/doc-plugin/src/index.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Application,
+  DeclarationReflection,
+  DefaultThemeRenderContext,
+  DefaultTheme,
+  Options,
+  JSX,
+  PageEvent,
+  Reflection,
+  ReflectionKind,
+  ReflectionGroup,
+} from 'typedoc';
+
+export class NavigationOverrideThemeContext extends DefaultThemeRenderContext {
+  constructor(theme: DefaultTheme, options: Options) {
+    super(theme, options);
+
+    this.navigation = (props) => {
+      return <>{overridePrimaryNavigation(this, props)}</>;
+    };
+  }
+}
+
+export class NavigationOverrideTheme extends DefaultTheme {
+  private _contextCache?: NavigationOverrideThemeContext;
+
+  override getRenderContext(): NavigationOverrideThemeContext {
+    this._contextCache ||= new NavigationOverrideThemeContext(
+      this,
+      this.application.options
+    );
+    return this._contextCache;
+  }
+}
+
+// Replicated from the TypeDoc codebase. This should be exported by TypeDoc.
+function classNames(
+  names: Record<string, boolean | null | undefined>,
+  extraCss?: string
+) {
+  const css = Object.keys(names)
+    .filter((key) => names[key])
+    .concat(extraCss || '')
+    .join(' ')
+    .trim()
+    .replace(/\s+/g, ' ');
+  return css.length ? css : undefined;
+}
+
+function overridePrimaryNavigation(
+  context: DefaultThemeRenderContext,
+  props: PageEvent<Reflection>
+) {
+  // TODO: We should filter out modules with <a href="_electron_forge_core._internal_.html"><internal></a>
+  const modules = props.model.project.getChildrenByKind(
+    ReflectionKind.SomeModule
+  );
+  const projectLinkName = modules.some((m) => m.kindOf(ReflectionKind.Module))
+    ? 'All Modules'
+    : 'Exports';
+
+  const groups = props.model.project.groups;
+
+  if (groups && groups.length !== 0) {
+    return (
+      <nav class="tsd-navigation primary">
+        <ul>
+          <li class={classNames({ current: props.model.isProject() })}>
+            <h3>
+              <a href={context.urlTo(props.model.project)}>{projectLinkName}</a>
+            </h3>
+          </li>
+          {groups.map(generateChildren)}
+        </ul>
+      </nav>
+    );
+  }
+
+  return (
+    <nav class="tsd-navigation primary">
+      <ul>
+        <li class={classNames({ current: props.model.isProject() })}>
+          <a href={context.urlTo(props.model.project)}>{projectLinkName}</a>
+        </li>
+      </ul>
+    </nav>
+  );
+
+  function generateChildren(group: ReflectionGroup) {
+    let childNav: JSX.Element | undefined;
+    if (group.children) {
+      const title = group.title;
+      if (group.children?.length) {
+        childNav = <ul>{group.children.map(link)}</ul>;
+      }
+
+      return (
+        <li
+          class={title + ' ' + classNames({ current: props.model.isProject() })}
```

Comment:
Clicking into here, I see a lot of `undefined` classes showing up in the HTML structure

e.g.

```html
<ul>
  <li class="@electron-forge/maker-appx undefined">
    <a href="_electron_forge_maker_appx.html">maker-appx</a>
  </li>
  <li class="@electron-forge/maker-deb undefined">
    <a href="_electron_forge_maker_deb.html">maker-deb</a>
  </li>
  <li class="@electron-forge/maker-dmg undefined">
    <a href="_electron_forge_maker_dmg.html">maker-dmg</a>
  </li>
  <li class="@electron-forge/maker-flatpak undefined">
    <a href="_electron_forge_maker_flatpak.html">maker-flatpak</a>
  </li>
  <li class="@electron-forge/maker-pkg undefined">
    <a href="_electron_forge_maker_pkg.html">maker-pkg</a>
  </li>
  <li class="@electron-forge/maker-rpm undefined">
    <a href="_electron_forge_maker_rpm.html">maker-rpm</a>
  </li>
  <li class="@electron-forge/maker-snap undefined">
    <a href="_electron_forge_maker_snap.html">maker-snap</a>
  </li>
  <li class="@electron-forge/maker-squirrel undefined">
    <a href="_electron_forge_maker_squirrel.html">maker-squirrel</a>
  </li>
  <li class="@electron-forge/maker-wix undefined">
    <a href="_electron_forge_maker_wix.html">maker-wix</a>
  </li>
  <li class="@electron-forge/maker-zip undefined">
    <a href="_electron_forge_maker_zip.html">maker-zip</a>
  </li>
</ul>;

```

Reply:
Ah, I think I know what that is - I'll clean that up, it's a holdover from how typescript generates classNames for CSS custom themes